templates/commands/goldentest/record.go

```diff
@@ -85,14 +88,33 @@ func (c *RecordCommand) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("failed to render golden tests: %w", merr)
 	}
 
-	// Render outputs in template golden test directories.
-	// TODO(chloechien): Copy rendered contents recursively instead of rendering twice.
 	logger := logging.FromContext(ctx)
+
+	// Recursively copy files from tempDir to template golden test directory.
 	for _, tc := range testCases {
-		if err := renderTestCase(c.flags.Location, c.flags.Location, tc); err != nil {
-			return fmt.Errorf("fatal error when recording golden test: %w", err)
+		testDir := filepath.Join(c.flags.Location, goldenTestDir, tc.TestName)
+		if err := clearTestDir(testDir); err != nil {
+			return fmt.Errorf("failed to clear test directory: %w", err)
+		}
+
+		visitor := func(relToAbsSrc string, de fs.DirEntry) (common.CopyHint, error) {
+			if !de.IsDir() {
+				logger.InfoContext(ctx, "recording", "testname", tc.TestName, "testdata", relToAbsSrc)
+			}
+			return common.CopyHint{
+				Overwrite: true,
```

Comment:
Is there an edge case here? What if the template coincidentally outputs a file named test.yaml? Would that conflict with the goldentest test.yaml?

It's not really specific to this PR, I just realized this while reviewing.

Reply:
We're adopting the simplest solution at the moment - reserve test.yaml same as spec.yaml.

Or another way is
testdata/golden/testname/test.yaml
testdata/golden/testname/data/....

WDYT?

Reply:
I think it's different than spec.yaml. spec.yaml is reserved in the template *source* directory, not in the output. The user can still output a file named spec.yaml if they want to, they'll just have to do something like `include renamed_spec.yaml as spec.yaml`. With test.yaml, it's different, because there's no workaround for a reserved filename in the template output, it's just impossible to output a ./test.yaml file.

Your idea for a `data` directory sounds good to me 👍

Reply:
Good catch in input/output difference! I'll update to data/ structure :)